ngraph.mk

```diff
@@ -145,40 +165,72 @@ $(NGRAPH_BRIDGE_OBJ): %.o: ngraph $(NGRAPH_BRIDGE_SRC)
   # nGraph provides some libraries that may compete with other libraries already installed
   # on the system. This provides the link-flags that, if provided early enough on the static-linker
   # command line, should ensure that the nGraph-supplied version is preferred.
+  ifeq ("$(shell uname)","Darwin")
+    NGRAPH_MKLML_LIB_ := mklml
+  else
+    NGRAPH_MKLML_LIB_ := mklml_intel
+  endif
+
   NGRAPH_COMMON_LIBRARY_LDFLAGS_ := \
     -ltbb \
     -liomp5 \
-    -lmklml_intel \
+    -l$(NGRAPH_MKLML_LIB_) \
     -lmkldnn
 
-  NGRAPH_LDFLAGS_ := \
-    -L$(MXNET_LIB_DIR) \
-    -Wl,-rpath-link=$(MXNET_LIB_DIR) \
-    $(NGRAPH_COMMON_LIBRARY_LDFLAGS_) \
-    -lngraph \
-    -lcpu_backend
-
-  ifeq ($(USE_NGRAPH_DISTRIBUTED), 1)
-    NGRAPH_LDFLAGS_ += $(shell mpicxx --showme:link)
+  ifeq ("$(shell uname)","Darwin")
+    NGRAPH_LDFLAGS_ := \
+      -L$(MXNET_LIB_DIR) \
+      $(NGRAPH_COMMON_LIBRARY_LDFLAGS_) \
+      -lngraph \
+      -lcpu_backend
+
+    ifeq ($(USE_NGRAPH_DISTRIBUTED), 1)
+      $(error "USE_NGRAPH_DISTRIBUTED=1 is not yet supported on OS X.")
+    endif
+
+    # The static-link-time flags for creating .so/.dylib files that will dynamically link to nGraph's libs
+    # at runtime.
+    NGRAPH_LDFLAGS_FOR_SHARED_LIBS := \
+      $(NGRAPH_LDFLAGS_) \
+      -Wl,-rpath,@loader_path
```

Comment:
Why no --as-needed?

Reply:
Short answer: It's not provided in Apple's version of `ld`.

Longer answer: I'm planning to audit the runtime binding of math libraries in a future PR.  But given how messy it got on Linux, I didn't want to let that hold up this current PR.